lib/msf/core/post/windows/process.rb

```diff
@@ -44,20 +44,28 @@ def get_notepad_pathname(bits, windir, client_arch)
   end
 
   #
-  # Injects a reflective DLL in to a process, and executes it.
+  # Injects a reflective DLL into a process, and executes it.
   #
   # @param rdll_path [String] The path to the DLL to inject
   # @param param     [String, Integer, nil] The parameter to pass to the DLL's entry point. If this value is a String
   #   then it will first be written into the process memory and then passed by reference. If the value is an Integer,
   #   then the value will be passed as is. If the value is nil, it'll be passed as a NULL pointer.
-  # @param pid       [Integer] The process ID to inject to, if unspecified, a new instance of notepad.exe will be
-  #   launched to host the injected DLL.
-  def execute_dll(rdll_path, param=nil, pid=nil)
+  # @param pid       [Integer] The process ID to inject to, if unspecified, a new instance of a random EXE from the
+  #   process_list array will be launched to host the injected DLL.
+  # @param is_wow64  [Boolean] If set, the process that is spawned needs to be WoW64 compatiable. Otherwise
+  #   spawn the default binary.
```

Comment:
Instead of using a boolean here, it may be better to use an ARCH constant. You'd then within your function infer that if the host environment is 64-bit (not the Meterpreter) then you'd use one of the SysWOW64 subfiles. If however the architecture is specified as 64-bit and the host is 32-bit you'd fail some how.

Reply:
Resolved now thanks to help from @smcintyre-r7 who suggested using the following code to try find the DLL properties dynamically using our existing Rex code:

```
pe = Rex::PeParsey::Pe.new_from_file(library_path, true)
pe.hdr.file['Machine'].value
```